integration/querydsl/expressions/src/main/java/com/blazebit/persistence/querydsl/BlazeCriteriaBuilderRenderer.java

```diff
@@ -1342,4 +1337,92 @@ public Object visit(TemplateExpression<?> templateExpression, Object o) {
             return (X) windowContainerBuilder;
         }
     }
+
+
+    /**
+     * Context for the {@link BindResolver}
+     *
+     * @since 1.6.0
+     */
+    private static class BindResolverContext {
+
+        private Path<?> alias;
+        private Path<?> cteAttribute;
+
+        public Path<?> getAlias() {
+            return alias;
+        }
+
+        public String getAliasString() {
+            return alias == null ? null : alias.getMetadata().getName();
+        }
+
+        public void setAlias(Path<?> alias) {
+            this.alias = alias;
+        }
+
+        public Path<?> getCteAttribute() {
+            return cteAttribute;
+        }
+
+        public void setCteAttribute(Path<?> cteAttribute) {
+            this.cteAttribute = cteAttribute;
+        }
+
+    }
+
+    /**
+     * Resolve the bind and alias information for an expression.
+     *
+     * @since 1.6.0
+     */
+    private static class BindResolver implements Visitor<Expression<?>, BindResolverContext> {
```

Comment:
I need to return three values here: the alias, the bind attribute path and the select expression. Should I integrate the Context and the Visitor? I've decided for now to keep the state seperate from the visitor.

Reply:
It's fine either way. Whatever you prefer.